miden-lib/asm/sat/internal/note.masm

```diff
@@ -68,21 +90,34 @@ end
 #!
 #! Inputs: []
 #! Outputs: []
-export.reset_current_consumed_note_ptr
-    # get the current consumed note index
-    exec.layout::get_current_consumed_note_idx
-    # => [note_idx, ...]
+export.note_processing_teardown
+    # set the current consumed note pointer to 0
+    push.0 exec.layout::set_current_consumed_note_ptr
+    # => []                                                                                                                                            
+end
+
+#! Prepares for the execution of a consumed note.  This involves:
+#! 1. Incrementing the current consumed note index and pointer.
+#! 2. Loading the note script root onto the stack.
+#!
+#! Stack: []
+#! Output: [NSR]
+#!
+#! - NSR is the note script root of the note currently being executed.
+export.prepare_note.4
+    # load the note index onto the stack
+    exec.increment_current_consumed_note_idx
+    # => [idx]
 
-    # if at least one note has been consumed we need to clear the note outputs from the stack
-    eq.0 not
-    # => [has_processed_notes, ...]
+    # convert the index of the consumed note being executed to a pointer and store in memory
+    exec.layout::get_consumed_note_ptr
+    # => [note_ptr]
 
-    if.true
-        dropw dropw dropw dropw
-    end
-    drop
-    # => []
+    # set current consumed note pointer to the note being executed
+    dup exec.layout::set_current_consumed_note_ptr
+    # => [note_ptr]
 
-    # set the current consumed note pointer to 0
-    push.0 exec.layout::set_current_consumed_note_ptr
+    # read the note script root onto the stack
+    exec.layout::get_consumed_note_script_root
+    # => [NOTE_SCRIPT_ROOT, i11, ..., i0]
 end
```

Comment:
> A few comments/questions here:
> 
> 1. Why do we need 4 local variables here? I'm not seeing them being used anywhere.

They're not needed anymore - they used to be required when we read the inputs from the advice provider using `adv_pipe` into local memory.  I will remove them.

> 2. In the comment on line 122, what are the `i11 ... i0` values?

These are referring to the note inputs however it is no longer accurate - I will remove them.

> 3. There is probably a reason for this which I've forgotten, but why do we track index and pointer of the currently active note separately? It seems like we could track just the index and compute the pointer via a simple formula. Something like `BASE_OFFSET + NOTE_WIDTH * i`. To signal that all notes have been processed, we could pick some invalid value for the index - e.g., `Felt::MAX`.

It looks like it was introduced as part of https://github.com/0xPolygonMiden/miden-base/pull/147.  I think we can migrate to exclusively storing the pointer in memory. The preference to store the pointer is because we have a number of methods that use the current note pointer and we would like to avoid them having to compute it every time.  The index is only ever used to update the pointer.  Shall I implement this change?

Reply:
I think storing just the pointer is fine. Not sure what the "impact area" of this change would be. If it is fairly small, we can do it in this PR. But if not, probably creating an issue and addressing in a subsequent PR is better.

Reply:
Created https://github.com/0xPolygonMiden/miden-base/issues/287 for removal of `CURRENT_CONSUMED_NOTE_IDX`